cli/mender-artifact/partition.go

```diff
@@ -62,18 +62,22 @@ func (v vFile) Open(comp artifact.Compressor, imgpath string) (VPFile, error) {
 	if err != nil {
 		return nil, err
 	}
-	modcands, isArtifact, err := getCandidatesForModify(imgname)
+	candidateType, modcands, err := getCandidatesForModify(imgname)
 	if err != nil {
 		return nil, err
 	}
-	for i := 0; i < len(modcands); i++ {
-		modcands[i].name = imgname
+	if modcands != nil {
```

Comment:
VirtualPartitionFiles will never modify artifact metadata like name, so couldn't the test here simply be if it does not contain any partitions. Error?

Reply:
Mmm, right. But then what about the last return?

Because it was bolean, if artifact, else; but now we have the enum, so if artifact, elif sdimg, else?

Reply:
Ok, I got a suggestion for this. Check the new change set

Reply:
Yes! Way better imo